### PR TITLE
feat(sql): MariaDB/MySQL syntax, Apache Hive syntax, unified builtin functions, floats, negative integers

### PR DESCRIPTION
**Pull request by LeoniePhiline**
# PR contents

All commits are **self-contained**, documenting

- upstream changes, 
- adjusted `languages.toml` git ref and 
- adjusted `runtime/queries/sql/highlights.scm` helix TS queries 

**per each feature**.

Therefore: 

**Each commit may be reviewed individually.** 

## MariaDB/MySQL table options `COLLATE`, `CHARACTER SET`, `ENGINE` (https://github.com/helix-editor/helix/commit/2b0ff71c385379a4d941b9ffc8871e29c5499ea8)

Upstream changes: https://github.com/DerekStride/tree-sitter-sql/compare/286e10c5bc5d1703ee8f9afb351165a9a6182be1...30e15d45dceb24ea51acf81ee7d75d81567b6e02

## Optional `COLUMN` keyword in `ALTER TABLE` (https://github.com/helix-editor/helix/pull/6041/commits/3c8b951ed35e94ab5376dd773537f6da35c4d394)

Upstream changes: https://github.com/DerekStride/tree-sitter-sql/compare/30e15d45dceb24ea51acf81ee7d75d81567b6e02...c508e6044adf4298d7b321f966c90cbe32d75d23

## Add `UNSIGNED` support, refactor numeric types (https://github.com/helix-editor/helix/pull/6041/commits/a564d8d006eab648b41bad2915854a668192b201)
    
Upstream changes: https://github.com/DerekStride/tree-sitter-sql/compare/c508e6044adf4298d7b321f966c90cbe32d75d23...2d1d5b68a1e11796dd0f4f068fc3e9d7e59fe9f7

## Add support for Apache Spark create Hive table (https://github.com/helix-editor/helix/pull/6041/commits/2951dccb50b2000409261eecb1385837e15c6602)

Upstream changes: https://github.com/DerekStride/tree-sitter-sql/compare/2d1d5b68a1e11796dd0f4f068fc3e9d7e59fe9f7...7be06f4d5eabace883dd45959c13dc740f1f1b98

## Add support for signed and unsigned floating point literals (https://github.com/helix-editor/helix/pull/6041/commits/51a1b94568014a834b32225b2e4c899d56dfde5c + https://github.com/helix-editor/helix/pull/6041/commits/4ab3a2252f3d3862eeafedf6d6128e03f1ce19a1)

Upstream changes: https://github.com/DerekStride/tree-sitter-sql/pull/92/files

## Add interval data type (https://github.com/helix-editor/helix/pull/6041/commits/4a960c7f5822c311e00b1958b4da4d81c0f94d9d)

Upstream changes: https://github.com/DerekStride/tree-sitter-sql/compare/13d375dea377bae5f235176fae97a50ba584db54...7b4bcd0394d759a660f470a4f07aa08b7b130d8c

## Add support for DROP INDEX (https://github.com/helix-editor/helix/pull/6041/commits/16549e2da994513ecee17c90ff5750f09230e8e2)

Upstream changes: https://github.com/DerekStride/tree-sitter-sql/compare/7b4bcd0394d759a660f470a4f07aa08b7b130d8c...173d6feb5064defb7d0ef742a4fc7c6d763a2df0

## Add MariaDB/MySQL `ALTER TABLE ... CHANGE|MODIFY ... [FIRST|AFTER]` syntax (https://github.com/helix-editor/helix/pull/6041/commits/4ea492fb975ab6cbea48b824150f70699bf52de5)

Upstream changes: https://github.com/DerekStride/tree-sitter-sql/compare/173d6feb5064defb7d0ef742a4fc7c6d763a2df0...0d7a121b2a08fb37109f7be1cc6654443cad661f

## Extract fields from Apache Hive storage location and row format (https://github.com/helix-editor/helix/pull/6041/commits/1ff0ebdcfe39428907429265e867af0c70afc12e)

Upstream changes: https://github.com/DerekStride/tree-sitter-sql/compare/0d7a121b2a08fb37109f7be1cc6654443cad661f...d2f0f6695fffa4ec1c81fc2060eddf83161f9ee3

## Fix unified built-in functions (https://github.com/helix-editor/helix/pull/6041/commits/4b30f1eecfb3801d9cb8016a3c8326b723df96f8)

Upstream changes: https://github.com/DerekStride/tree-sitter-sql/compare/d2f0f6695fffa4ec1c81fc2060eddf83161f9ee3...e4e43ba742a2ee88cbb24dbf305a7daadd583873

## Support negative integers (https://github.com/helix-editor/helix/pull/6041/commits/afbe26ac19f2e62bd37e41c1572cd1f39f4e0efb)

Upstream changes: https://github.com/DerekStride/tree-sitter-sql/compare/e4e43ba742a2ee88cbb24dbf305a7daadd583873...3a3f92b29c880488a08bc2baaf1aca6432ec3380

### CODE REVIEW

**Comment by LeoniePhiline** [comment]
Ready for review.

File: runtime/queries/sql/highlights.scm [hunk 1]
```diff
@@ -57,7 +57,7 @@
 ] @constant.builtin
 
 ((literal) @constant.numeric
-  (#match? @constant.numeric "^(-?\d*\.?\d*)$"))
+  (#match? @constant.numeric "^-?[0-9]*\.?[0-9]*$"))
```
Comment (archseer) [inline runtime/queries/sql/highlights.scm:57]: Why the change here? \d seems simpler

Comment (LeoniePhiline) [reply runtime/queries/sql/highlights.scm:57]: `\d` is apparently not supported. `\d` won't match where `[0-9]` matches.

Comment (archseer) [reply runtime/queries/sql/highlights.scm:57]: It's using the rust regex engine and it should be supported

Comment (LeoniePhiline) [reply runtime/queries/sql/highlights.scm:57]: Alright. Is it an escaping problem?  I'll give it another quick try...

Comment (LeoniePhiline) [reply runtime/queries/sql/highlights.scm:57]: @archseer Here you go:

Current version (correct highlighting):

Screenshot_20230219_173609

Adjusted version, then `:config-reload` (incorrect highlighting of numbers as strings):

Screenshot_20230219_173720

Comment (LeoniePhiline) [reply runtime/queries/sql/highlights.scm:57]: Now surprisingly, this works:

grafik

How was the `\.` not a problem when `\\.` would have been required?

**Review by dead10ck — APPROVED** [review]
This seems fine to me

**Review by the-mikedavis — APPROVED** [review]
